modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -258,12 +263,6 @@ private void initConfigurationResolver() {
     // check for implicit provider
     findConfigurationProviderResolver().ifPresent(configurationProviderResolver::set);
 
-    Optional<ConfigurationInstance> staticConfiguration = getStaticConfiguration();
-    if (staticConfiguration.isPresent()) {
-      configurationResolver = event -> staticConfiguration;
-      return;
-    }
-
```

Comment:
This part is being removed because it had actually no effect. The method [getStaticConfiguration](https://github.com/mulesoft/mule/blob/2a87abccb12f634037000cc782b1e46318fa9b2f/modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java#L489) will always return empty at this point because the `configurationResolver` is still `null`.

While removing it could be considered out of scope for this PR, leaving it would complicate things, because we are trying to cache the result, and we don't want it to be always empty. I could use the internal, uncached, `doGetStaticConfiguration` but that will just add more to the confusion.

We could consider trying to actually fix this so that this optimization attempt actually works. But in the end, it is probably not a big win vs calling resolveConfigFromProvider which will just do a get on the provider. Considering the optimization was only meant for static ConfigurationProviders, that should be a direct return of a cached value. Notice that here we don't deal with the Null Event cost because the real event is given to the resolver.